src/rust/madara/template/src/template.rs

```diff
@@ -0,0 +1,344 @@
+			+ "/" + manga_id.as_str()
+			+ "/ajax/chapters";
+	}
+
+	let int_id = get_int_manga_id(manga_id, data.base_url.clone(), data.source_path.clone());
+	let body_content = format!("action=manga_get_chapters&manga={}", int_id);
+
+	let req = Request::new(url.clone().as_str(), HttpMethod::Post)
+		.body(body_content.as_bytes())
+		.header("Content-Type", "application/x-www-form-urlencoded");
+
+	let html = req.html();
+
+	let mut chapters: Vec<Chapter> = Vec::new();
+	for item in html.select("li.wp-manga-chapter  ").array() {
+		let obj = item.as_node();
+
+		let id = obj
+			.select("a")
+			.attr("href")
+			.read()
+			.replace(&(data.base_url.clone() + "/"), "")
+			.replace(&(data.source_path.clone() + "/"), "");
+
+		let mut title = String::new();
+		let t_tag = obj.select("a").text().read();
+		if t_tag.contains("-") {
+			title.push_str(t_tag[t_tag.find("-").unwrap() + 1..].trim());
+		}
+
+		/*  Chapter number is first occourance of a number in the last element of url
+			when split with "/"
+			e.g.
+			one-piece-color-jk-english/volume-20-showdown-at-alubarna/chapter-177-30-million-vs-81-million/
+			will return 177
+			parasite-chromatique-french/volume-10/chapitre-062/
+			will return 62
+		*/
+		let mut chapter = 0.0;
+
+		let slash_vec = id.as_str().split("/").collect::<Vec<&str>>();
+
+		let dash_split = slash_vec[slash_vec.len() - 2].split("-");
+		let dash_vec = dash_split.collect::<Vec<&str>>();
+
+		for obj in dash_vec {
+			let item = obj.replace("/", "").parse::<f32>().unwrap_or(-1.0);
+			if item != -1.0 {
+				chapter = item;
+				break;
+			}
+		}
+
+		let date_str = obj.select("span.chapter-release-date > i").text().read();
+		let mut date_updated = StringRef::from(&date_str)
+			.0
+			.as_date("MMM d, yyyy", Some("en"), None)
+			.unwrap_or(-1.0);
+		if date_updated == -1.0 {
+			date_updated = current_date();
+		}
+
+		let url = obj.select("a").attr("href").read();
+		let lang = data.lang.clone();
+
+		chapters.push(Chapter {
+			id,
+			title,
+			volume: -1.0,
+			chapter,
+			date_updated,
+			scanlator: String::new(),
+			url,
+			lang,
+		});
+	}
+	Ok(chapters)
+}
+
+pub fn get_page_list(chapter_id: String, data: MadaraSiteData) -> Result<Vec<Page>> {
+	let url =
+		data.base_url.clone() + "/" + data.source_path.clone().as_str() + "/" + chapter_id.as_str();
+	let html = Request::new(url.clone().as_str(), HttpMethod::Get).html();
+
+	let mut ind = 0;
+	let mut pages: Vec<Page> = Vec::new();
+	for item in html.select(data.image_selector.clone().as_str()).array() {
+		pages.push(Page {
+			index: ind,
+			url: get_image_url(item.as_node()),
+			base64: String::new(),
+			text: String::new(),
+		});
+		ind += 1;
+	}
+	Ok(pages)
+}
+
+pub fn modify_image_request(base_url: String, request: Request) {
+	request.header("Referer", &base_url);
+}
+
+pub fn handle_url(url: String, data: MadaraSiteData) -> Result<DeepLink> {
+	Ok(DeepLink {
+		manga: Some(get_manga_details(url.clone(), data)?),
```

Comment:
Don't think this works. `get_manga_details` takes a manga id, not a url, so you need to parse out the manga id from the url. Also, you should consider how chapter urls are handled in addition to manga urls, even if you don't return the chapter.